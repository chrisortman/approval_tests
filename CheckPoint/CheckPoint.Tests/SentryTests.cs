using ApprovalTests;
using NUnit.Framework;
using TypeMock.ArrangeActAssert;

namespace CheckPoint.Tests
{
	[TestFixture]
	public class SentryTests
	{
		[Test]
		[Isolated]
		public void GetInteractions()
		{
			Sentry s = Monitor.Interactions(typeof (Controller));

			var controller = new Controller();
			controller.Action();

			Approvals.Approve(s.Report);
		}

		[Test]
		[Isolated]
		public void GetInteractionsAgainstSystemTypes()
		{
			Sentry s = Monitor.Interactions(typeof (Controller));

			var controller = new Controller();
			controller.ActionsAgainstSystemTypes();

			Approvals.Approve(s.Report);
		}

		[Test]
		[Isolated]
		public void GetListOfCalls()
		{
			Sentry s = Monitor.Calls<Model>();

			var controller = new Controller();
			controller.Action();

			Approvals.Approve(s.Report);
		}
	}
}