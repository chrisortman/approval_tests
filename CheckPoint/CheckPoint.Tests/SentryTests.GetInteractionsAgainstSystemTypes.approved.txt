(CheckPoint.Tests.Model) model..ctor()
(CheckPoint.Tests.Model) <Model2>k__BackingField..ctor()
(CheckPoint.Tests.Model) model.ReturnsSystemType()
(CheckPoint.Tests.Model) model.ReturnsInt()
(CheckPoint.Tests.Model) model.get_SomeProperty()
(CheckPoint.Tests.Model) <Model2>k__BackingField.ReturnsSystemType()
(CheckPoint.Tests.Model) <Model2>k__BackingField.ReturnsInt()
(CheckPoint.Tests.Model) model.get_SomeProperty()
